pkg/vsphere/disk/disk_manager.go

```diff
@@ -310,6 +315,9 @@ func (m *Manager) Detach(op trace.Operation, d *VirtualDisk) error {
 	}
 
 	if err := d.canBeDetached(); err != nil {
+		// even though canBeDetached() is called here and nowhere else, it does not imply
+		// an attempt to detach, so decrease the ref count from here for that reason
+		d.attachedRefs--
```

Comment:
`canBeDetached` will return an error if the disk is mounted, the disk is already detached, or the disk is in use by someone else (either mounted or attached). So basically, this client attempted to detach the disk because it no longer needs it, but the disk is in use elsewhere and thus can't be detached yet, but the reference count needs to be updated to show one less consumer using the disk.